app/assets/javascripts/discourse-markdown-it/src/features/bbcode-block.js

```diff
@@ -256,10 +247,7 @@ function applyBBCode(state, startLine, endLine, silent, md) {
     if (startLine === nextLine) {
       content = state.src.slice(start + info.length, closeTag.start);
     } else {
-      content = state.src.slice(
-        state.bMarks[startLine + 1],
-        state.eMarks[nextLine - 1]
-      );
+      content = state.getLines(startLine + 1, nextLine, 0, false);
```

Comment:
This line is the actual bug fix. Doing a `start.src.slice` was not taking into account being nested inside a block quote for example (which prepends a `> ` on every line).

The `getLines` function does all the heavy lifting for us 👏